src/validators.ts

```diff
@@ -165,3 +165,28 @@ export function map<K, V>(
     return result;
   };
 }
+
+export function either<F, S>(
+  first: Validator<F>,
+  second: Validator<S>,
+): Validator<F | S> {
+  return (val: any): F | S => {
+    let result: F | S;
+
+    try {
+      result = first(val);
+    } catch (_) {
+      try {
+        result = second(val);
+      } catch (e) {
+        throw new Error(
+          `Expected an ${first.name} or ${second.name} but received ${repr(
```

Comment:
`Function.name()` doesn't work in production as noted in this post on StackOverflow - [link](https://stackoverflow.com/a/61326623/10221282) - it will resolve in `""`

Reply:
Only in the presence of minification. I think that's fine.